server/api/middlewares/validateClientSecurity.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response, NextFunction } from 'express';
+import { validateS2sToken } from '../services/s2sTokenValidation.service';
+import { validateUserToken } from '../services/userTokenValidation.service';
+import * as process from 'process';
+
+export async function validateIncomingRequest(req: Request, res: Response, next: NextFunction) {
+    let clientServiceName;
+    const urlS2s = process.env.S2S_TOKEN_URL;
+    let rawS2sToken: string | string[] | undefined;
+    let s2sToken: string | string[] | undefined;
+
+    // Check if authorisation header exists, if not add dummy value
+    if (req.headers.serviceauthorization) {
+        rawS2sToken = String(req.headers.serviceauthorization);
```

Comment:
if the header can be `string`, `string[]`, or `undefined`, this is the place where we have to deal with that.

if it's `string`, we can simply use it as is.
if it's `string[]`, we have to decide if we need to use the first or the last value (probably last, as identical headers are meant to be overwriting previous ones, unless specifically stated otherwise)
if it's `undefined`, it means the header wasn't set, and we should probably bail out at this point.

By forcing all 3 of those cases into a string, you will get weird side-effects, such as ending up with headers like `Bearer undefined` (when you force undefined to become a string). If it was an array, you end up with a string build up of all array values, comma separated. Probably also not what should be used.